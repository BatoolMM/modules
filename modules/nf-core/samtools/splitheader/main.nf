process SAMTOOLS_SPLITHEADER {
    tag "$meta.id"
    label 'process_single'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/samtools:1.22.1--h96c455f_0' :
        'biocontainers/samtools:1.22.1--h96c455f_0' }"

    input:
    tuple val(meta), path(input)

    output:
    tuple val(meta), path("*_readgroups.txt"), emit: readgroup
    tuple val(meta), path("*_programs.txt")  , emit: programs
    tuple val(meta), path("*_sequences.txt") , emit: sequences
    path  "versions.yml"                     , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args   = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    samtools \\
        view \\
        -H \\
        $args \\
        $input \\
    | tee \\
        >( grep '^@RG' > ${prefix}_readgroups.txt ) \
        >( grep '^@PG' > ${prefix}_programs.txt ) \
        >( grep '^@SQ' > ${prefix}_sequences.txt ) \
     > /dev/null

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """

    stub:
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    echo -e "@RG\\tID:${prefix}\\tSM:${prefix}\\tPL:ILLUMINA" > ${prefix}_readgroups.txt
    echo -e "@PG\\tID:samtools.4\\tPN:samtools\\tPP:samtools.3\\tVN:1.22.1\\tCL:samtools view -H ${input}" > ${prefix}_programs.txt
    echo -e "@SQ\\tSN:chr1\\tLN:10000" > ${prefix}_sequences.txt

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        samtools: \$(echo \$(samtools --version 2>&1) | sed 's/^.*samtools //; s/Using.*\$//')
    END_VERSIONS
    """
}
