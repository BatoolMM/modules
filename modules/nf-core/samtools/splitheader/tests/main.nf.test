
nextflow_process {

    name "Test Process SAMTOOLS_SPLITHEADER"
    script "../main.nf"
    process "SAMTOOLS_SPLITHEADER"

    tag "modules"
    tag "modules_nfcore"
    tag "samtools"
    tag "samtools/splitheader"

    test("test-samtools-splitheader") {

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.bam', checkIfExists: true)
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
					file(process.out.readgroup[0][1]).readLines(),
					file(process.out.programs[0][1]).readLines(),
					file(process.out.sequences[0][1]).readLines(),
					process.out.versions
					).match()
				}
            )
        }
    }

    test("test-samtools-splitheader - stub") {

        options "-stub"

        when {
            process {
                """
                input[0] = [
				    [ id:'test', single_end:false ], // meta map
				    file(params.modules_testdata_base_path + 'genomics/sarscov2/illumina/bam/test.paired_end.bam', checkIfExists: true)
				]

                """
            }
        }

        then {
            assertAll(
                { assert process.success },
                { assert snapshot(
					file(process.out.readgroup[0][1]).readLines(),
					file(process.out.programs[0][1]).readLines(),
					file(process.out.sequences[0][1]).readLines(),
					process.out.versions
					).match()
				}
            )
        }
    }

}
