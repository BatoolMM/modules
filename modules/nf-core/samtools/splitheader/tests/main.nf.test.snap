{
    "test-samtools-splitheader": {
        "content": [
            [
                "@RG\tID:1\tLB:lib1\tPL:ILLUMINA\tSM:test\tPU:barcode1"
            ],
            [
                "@PG\tID:minimap2\tPN:minimap2\tVN:2.17-r941\tCL:minimap2 -ax sr tests/data/fasta/sarscov2/GCA_011545545.1_ASM1154554v1_genomic.fna tests/data/fastq/dna/sarscov2_1.fastq.gz tests/data/fastq/dna/sarscov2_2.fastq.gz",
                "@PG\tID:samtools\tPN:samtools\tPP:minimap2\tVN:1.11\tCL:samtools view -Sb sarscov2_aln.sam",
                "@PG\tID:samtools.1\tPN:samtools\tPP:samtools\tVN:1.22.1\tCL:samtools view -H test.paired_end.bam"
            ],
            [
                "@SQ\tSN:MT192765.1\tLN:29829"
            ],
            [
                "versions.yml:md5,1c199f83f54556a51c500ed49ea6102a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-27T10:16:40.848026"
    },
    "test-samtools-splitheader - stub": {
        "content": [
            [
                "@RG\tID:test\tSM:test\tPL:ILLUMINA"
            ],
            [
                "@PG\tID:samtools.4\tPN:samtools\tPP:samtools.3\tVN:1.22.1\tCL:samtools view -H test.paired_end.bam"
            ],
            [
                "@SQ\tSN:chr1\tLN:10000"
            ],
            [
                "versions.yml:md5,1c199f83f54556a51c500ed49ea6102a"
            ]
        ],
        "meta": {
            "nf-test": "0.9.3",
            "nextflow": "25.04.2"
        },
        "timestamp": "2025-10-27T10:16:43.794756"
    }
}